qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -174,30 +179,35 @@ def show_preview(self):
 
     def download_resource(self):
         resource = self.selected_resource()
-        file_path = download_resource_file(
-            resource.file, resource.resource_type, resource.uuid
-        )
+        file_extension = os.path.splitext(resource.file)[1]
+
+        file_path = QFileDialog.getSaveFileName(
+            self, "Save Resource", "", f"All Files (*{file_extension})"
+        )[0]
+
         if file_path:
-            text = f"Successfully download {resource.name} to {file_path}"
+            full_file_path = file_path + file_extension
+            download_resource_file(
+                resource.file, resource.resource_type, resource.uuid, full_file_path
+            )
+            text = "Successfully downloaded {} to {}".format(
```

Comment:
put this text in a `self.tr()`, so that the translation can pick it up. Also, it is a good practice to use a dictionary, e.g. 
```
"My name is {fname}, I'm {age}".format(fname = "John", age = 36)
```